dias/analyzers/source_spectra_analyzer.py

```diff
@@ -0,0 +1,733 @@
+                    index_map.create_dataset("pol", data=np.string_(pols))
+                    index_map.create_dataset("time", data=data.time)
+                    index_map.create_dataset("ra", data=ra)
+                    index_map.create_dataset("ha", data=ha)
+
+                    dset = handler.create_dataset("vis", data=vis)
+                    dset.attrs["axis"] = np.array(["freq", "pol", "ha"], dtype="S")
+
+                    dset = handler.create_dataset(
+                        "weight", data=tools.invert_no_zero(var)
+                    )
+                    dset.attrs["axis"] = np.array(["freq", "pol", "ha"], dtype="S")
+
+                    dset = handler.create_dataset("count", data=counter.astype(np.int))
+                    dset.attrs["axis"] = np.array(["freq", "pol", "ha"], dtype="S")
+
+                    if self.perform_fit:
+
+                        index_map.create_dataset("param", data="model_fitting")
+
+                        dset = handler.create_dataset("residual_noise", data=resid_rms)
+                        dset.attrs["axis"] = np.array(["freq", "pol"], dtype="S")
+
+                        dset = handler.create_dataset("parameter", data=parameter)
+                        dset.attrs["axis"] = np.array(
+                            ["freq", "pol", "param"], dtype="S"
+                        )
+
+                        dset = handler.create_dataset(
+                            "parameter_cov", data=parameter_cov
+                        )
+                        dset.attrs["axis"] = np.array(
+                            ["freq", "pol", "param", "param"], dtype="S"
+                        )
+
+                    handler.attrs["source"] = src
+                    handler.attrs["csd"] = csd
+                    handler.attrs["instrument_name"] = self.correlator
+                    handler.attrs["collection_server"] = subprocess.check_output(
+                        ["hostname"]
+                    ).strip()
+                    handler.attrs["system_user"] = subprocess.check_output(
+                        ["id", "-u", "-n"]
+                    ).strip()
+                    handler.attrs["git_version_tag"] = dias_version_tag
+
+                self.logger.info("File for {} successfully written out.".format(src))
+                self.run_metric.labels(source=src).inc()
+
+        if err_msg:
+            raise exception.DiasDataError(err_msg)
+
+    def get_baselines(self, indexmap, inputmap):
+        """Return baseline indices for averaging."""
+        prod = defaultdict(list)
+        prodmap = defaultdict(list)
+        dist = defaultdict(list)
+        conj = defaultdict(list)
+        cyl = defaultdict(list)
+        scale = defaultdict(list)
+
+        # Compute feed positions without rotation
+        tools.change_chime_location(rotation=self.rot)
+        feedpos = tools.get_feed_positions(inputmap)
+
```

Comment:
Before looping over `indexmap["stack"]`, we should redefine the stack axis so that it only uses inputs that are connected to CHIME antennas (as described by the`inputmap`).  This can be done with:

```
stack, stack_flag = tools.redefine_stack_index_map(inputmap, index_map["prod"], index_map["stack"], reverse_stack)
if not np.all(stack_flag):
    self.log.warning(
                "There are %d stacked baselines that are masked "
                "by the inputmap." % np.sum(~stack_flag)
            )
```

Note that this requires passing `reverse_stack` as an argument to `get_baselines`.  The `reverse_stack` is contained in `data.reverse_map["stack"]`.

Reply:
The for loop would then be:

```
for pp, (this_prod, this_conj) in enumerate(stack):
```

Reply:
Thanks Seth! I think we will also need to do this for sensitivity analyzer. I will do that fix as well.